### ISSUE TITLE
Difference between paper and code

### ISSUE THREAD

**Issue by fredo838** [pfnet-research/sngan_projection#18]
In the paper of Spectral Normalization, it is said that all experiments were done using the original GAN value function, namely: log(D(x) + log(1-D(G(x)), but in the implementation, in pfnet-research/sngan_projection/updater.py , the dcgan loss is defined as follows:

def loss_dcgan_dis(dis_fake, dis_real):
    L1 = F.mean(F.softplus(-dis_real))
    L2 = F.mean(F.softplus(dis_fake))
    loss = L1 + L2
    return loss

def loss_dcgan_gen(dis_fake):
    loss = F.mean(F.softplus(-dis_fake))
return loss

Are these the loss functions used for the experiments? If so, how does this relate to the paper (because this is basically the Wasserstein loss)? If not, where are the ones that the paper used located (in which case I'm just confused by this definition)?

**Comment by zhangqianhui** [comment]
'we are using the modified generator updated rule proposed by Goodfellow et al (2014), which uses softplus function -log sigmoid(f(x)) = log  (1 + exp(-f(x))) := softplus(-f(x)) in place of log (1- sigmoid(f(x)))’

The details can be found in https://openreview.net/forum?id=B1QRgziT-&noteId=B1QRgziT-

**Comment by takerum** [comment]
Hi, 
As @zhangqianhui cited our response from the openreview, `-log sigmoid(f(x)) = log (1 + exp(-f(x))) = softplus(-f(x))`  and `-log(1-sigmoid(f(x))) = log (1 + exp(f(x))) = softplus(f(x))`.
Thus, `softplus(-f(x)) +  softplus(f(x))` represents the same objective as the original adversarial loss function.
